spotinst/resource_spotinst_ocean_spark.go

```diff
@@ -217,5 +228,56 @@ func deleteSparkCluster(ctx context.Context, resourceData *schema.ResourceData,
 		return fmt.Errorf("[ERROR] onDelete() -> Failed to delete cluster: %s", err)
 	}
 
+	if forceDelete {
+		log.Printf("===> Cluster has been force deleted: %s <===", resourceData.Id())
+		return nil
+	}
+
+	if err := waitUntilClusterDeleted(ctx, resourceData, meta); err != nil {
+		return err
+	}
+
 	return nil
 }
+
+func waitUntilClusterDeleted(ctx context.Context, resourceData *schema.ResourceData, meta interface{}) error {
```

Comment:
Perhaps cast `meta` to a client beforehand and pass that around into the private methods.

```golang
oceanSparkClient := meta.(*Client).ocean.Spark()
```